### PR TITLE
Customer contribution features

### PR DESCRIPTION
**Pull request by daniel-yh**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including READMEs and comments (where appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs [hunk 1]
```diff
@@ -111,18 +118,54 @@ public void SetupConnection(bool console = false)
                 try
                 {
                     var file1 = httpService.DownloadS3FileAsync("newtonsoft.json.json.gz");
```
Comment (saikiranakula-amzn) [inline packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:120]: move the hardcoded strings to constants file

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs [hunk 2]
```diff
@@ -111,18 +118,54 @@ public void SetupConnection(bool console = false)
...
                 }
             });
+
+            _connection.On<CustomerFeedbackRequest, Response<bool, string>>("sendCustomerFeedback", request =>
+            {
+                try
+                {
+                    string endPoint = _cccongif.CustomerFeedbackEndpoint;
+                    string uniqueMachineID = LogUploadUtils.getUniqueIdentifier();
+                    string key = $"{uniqueMachineID}/{request.Date}";
```
Comment (saikiranakula-amzn) [inline packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:138]: is there any specific format we need to use for date to be consistent all the time. Refer this https://www.c-sharpcorner.com/blogs/date-and-time-format-in-c-sharp-programming1

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs [hunk 3]
```diff
@@ -18,12 +21,16 @@ public class Application
         private IServiceProvider _services { get; set; }
         private Connection _connection;
         private ILogger _logger;
+        private CustomerContributionConfiguration _cccongif;
```
Comment (saikiranakula-amzn) [inline packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:24]: typo _ccconfig

**Review by saikiranakula-amzn — CHANGES_REQUESTED** [review]

File: packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs [hunk 4]
```diff
@@ -19,9 +20,9 @@ public class MetricsBase
         public string TimeStamp { get; set; }
 
         // This identifier will be used as default, if there are no active network interface adapters on the machine
-        private const string DefaultIdentifier = "591E6A97031144D5BADCE980EE3E51B7";
-        /// <summary>
-        /// This property uniquely identifies the customers using porting assistant 
+        public const string DefaultIdentifier = "591E6A97031144D5BADCE980EE3E51B7";
```
Comment (saikiranakula-amzn) [inline packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs:23]: make the access modifier internal

**Review by saikiranakula-amzn — CHANGES_REQUESTED** [review]

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (cslong) [reply packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:24]: noticed that Sai already made a comment on this, so you can ignore this.

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs (hunk 1 above)
Comment (daniel-yh) [reply packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:120]: Hi Sai, it is from original code. I will let original author know this. Resolve it for now. Thanks!

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs (hunk 2 above)
Comment (daniel-yh) [reply packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:138]: I amended codes in front-end side (packages/react/src/components/CustomerContribution/CustomerFeedbackModal.tsx, line 64) to use ISO format date (such as 2021-10-05T14:48:00.000Z).

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs (hunk 3 above)
Comment (daniel-yh) [reply packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs]: Corrected!

File: packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs (hunk 4 above)
Comment (daniel-yh) [reply packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs]: Corrected.

**Comment by daniel-yh** [comment]
Somehow the code build failed. I'll close it and open new PR.

**Pull request closed without merging**